Assets/Tests/InputSystem/Plugins/UITests.cs

```diff
@@ -4016,6 +4016,260 @@ public IEnumerator UI_DisplayIndexMatchesDisplayMultiplePointers()
         Assert.AreEqual(0, scene.parentReceiver.events.Count);
     }
 
+    void ClickButton(Button button)
+    {
+        bool called = false;
+        button.onClick.AddListener(() => { called = true; });
+        button.OnPointerClick(new PointerEventData(EventSystem.current) { button = PointerEventData.InputButton.Left });
+        //Debug.Log(called);
+        Assert.AreEqual(called, true);
+    }
+
+    void ChangeCanvasRenderMode(Canvas canvas, RenderMode mode)
+    {
+        canvas.renderMode = mode;
+    }
+
+    [Test]
+    public void ButtonClickTest()
+    {
+        var allParent = GameObject.Instantiate(Resources.Load("AllParent"));
+
+        var go = GameObject.Find("Left Eye Canvas Screen Camera").transform.Find("Left Button");
+        Button leftButton = go.GetComponent<Button>();
+        Assert.IsNotNull(leftButton, "Can not find Left Button!");
+
+        Button rightButton = GameObject.Find("Right Eye Canvas Screen Camera").transform.Find("Right Button").GetComponent<Button>();
+        Assert.IsNotNull(leftButton, "Can not find Right Button!");
+
+        Canvas leftEyeCanvas = GameObject.Find("Left Eye Canvas Screen Overlay").GetComponent<Canvas>();
+        Assert.IsNotNull(leftEyeCanvas, "Can not find Left Eye Canvas Screen Overlay!");
+
+        Canvas rightEyeCanvas = GameObject.Find("Right Eye Canvas Screen Overlay").GetComponent<Canvas>();
+        Assert.IsNotNull(rightEyeCanvas, "Can not find Right Eye Canvas Screen Overlay!");
+
+        //ClickBothButtonOnCanvasOverlayMode
+        ChangeCanvasRenderMode(leftEyeCanvas, RenderMode.ScreenSpaceOverlay);
+        ChangeCanvasRenderMode(rightEyeCanvas, RenderMode.ScreenSpaceOverlay);
+        ClickButton(leftButton);
+        ClickButton(rightButton);
+
+        //ClickBothButtonOnCanvasCameraMode
+        ChangeCanvasRenderMode(leftEyeCanvas, RenderMode.ScreenSpaceCamera);
+        ChangeCanvasRenderMode(rightEyeCanvas, RenderMode.ScreenSpaceCamera);
+        ClickButton(leftButton);
+        ClickButton(rightButton);
+
+        //ClickBothButtonOnCanvasOverlayMode&CameraMode
+        ChangeCanvasRenderMode(leftEyeCanvas, RenderMode.ScreenSpaceOverlay);
+        ChangeCanvasRenderMode(rightEyeCanvas, RenderMode.ScreenSpaceCamera);
+        ClickButton(leftButton);
+        ClickButton(rightButton);
+
+        GameObject.DestroyImmediate(allParent);
+    }
+
+    [UnityTest]
+    public IEnumerator DragImageTest()
+    {
+        var playAllParent = GameObject.Instantiate(Resources.Load("PlayAllParent"));
```

Comment:
Instead of loading the prefab from the resources directory, we can use the InputSystem:

```
// Since you're using a mouse, we add a mouse to the input system
var mouse = InputSystem.AddDevice<Mouse>();

// this creates a test scene with a canvas set to screenspace and a left/right child
var scene = CreateTestUI();

// this create an InputActionAsset and populates the action map to allow you to "position" and "click" the mouse
var uiActions = actions.AddActionMap("UI");
var pointAction = uiActions.AddAction("point", type: InputActionType.PassThrough, binding: "<Mouse>/position");
var clickAction = uiActions.AddAction("click", type: InputActionType.PassThrough, binding: "<Mouse>/leftButton");
        actions.Enable();
scene.uiModule.point = InputActionReference.Create(pointAction);
scene.uiModule.leftClick = InputActionReference.Create(clickAction);
```

Reply:
Check out UI_DisplayIndexMatchesDisplayWithMouseOnOverlayCanvas to see a full example